core/Network/TLS/Handshake/State.hs

```diff
@@ -49,6 +50,7 @@ module Network.TLS.Handshake.State
     , getHandshakeMessages
     , getHandshakeMessagesRev
     , getHandshakeDigest
+    , foldHandshakeDigest
```

Comment:
This is used only in the `State13` module. Does it make sense to move this into `State13`?

Reply:
It's precisely what I try to avoid. Have all functions changing things like `hstHandshakeMessages` and `hstHandshakeDigest` at one place. And less 13-specific code.

Reply:
OK.